test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -288,3 +266,100 @@ func TestVLANNetwork(t *testing.T) {
 	}
 	testSecondaryNetwork(t, networkTypeVLAN, pods)
 }
+
+func (data *testData) assignIP() error {
+	e2eTestData := data.e2eTestData
+	namespace := e2eTestData.GetTestNamespace()
+
+	for _, testPod := range data.pods {
+		_, err := e2eTestData.PodWaitFor(defaultTimeout, testPod.podName, namespace, func(pod *corev1.Pod) (bool, error) {
+			if pod.Status.Phase != corev1.PodRunning {
+				return false, nil
+			}
+			podIPs, err := data.listPodIPs(testPod)
+			if err != nil {
+				return false, err
+			}
+			ip, exists := podIPs["eth1"]
+			if !exists || ip == nil {
+				logs.Infof("IP not available for interface 'eth1' in Pod %s, retrying...", testPod.podName)
+				return false, nil
+			}
+
+			if err := aws.AssignIPToEC2ENI(testPod.secondaryENI, string(ip)); err != nil {
+				return false, err
+			}
+			logs.Infof("assigned private ip addresses %s to interface %s", ip, testPod.secondaryENI)
+			return true, nil
+		})
+		if err != nil {
+			return fmt.Errorf("error when waiting for secondary IPs for Pod %+v: %v", testPod, err)
+		}
+	}
+	return nil
+}
+
+func TestSRIOVNetwork(t *testing.T) {
+	e2eTestData, err := antreae2e.SetupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer antreae2e.TeardownTest(t, e2eTestData)
+
+	awsTestData, err := setupTest(e2eTestData)
+	if err != nil {
+		t.Fatalf("Error when setting up AWS test data: %v", err)
+	}
+
+	node0Name := antreae2e.NodeName(0)
+	node1Name := antreae2e.NodeName(1)
+	node0ENI, err := awsTestData.retrieveNodeENI(node0Name)
+	if err != nil {
+		t.Fatalf("Error when retrieve secondary NetworkInf ID of Node %s: %v", node0Name, err)
+	}
+	node1ENI, err := awsTestData.retrieveNodeENI(node1Name)
+	if err != nil {
+		t.Fatalf("Error when retrieve secondary NetworkInf ID of Node %s: %v", node1Name, err)
+	}
```

Comment:
can we move this to the `assignIP` function and remove the `secondaryENI` field from `testPodInfo`? I don't think there is any consumer of `secondaryENI` besides the `assignIP` function?
this way, only the `assignIP` function would be "provider-dependent"

Reply:
done